schemas/codify.xsd

```diff
@@ -1,50 +1,220 @@
+          <xs:attributeGroup ref="lib:targetAttributes"/>
+          <xs:attributeGroup ref="sharedAttributes"/>
+          <xs:attribute name="after" type="xs:string" use="optional"/>
+          <xs:attribute name="before" type="xs:string" use="optional"/>
+          <xs:attribute name="heading-value" type="xs:string" use="optional"/>
+          <xs:attribute name="num-value" type="xs:string" use="optional"/>
+          <xs:attribute name="prefix-value" type="xs:string" use="optional"/>
+          <xs:attribute name="tag" type="xs:string" use="optional"/>
+          <xs:attribute name="technical" type="xs:string" use="optional"/>
+        </xs:extension>
+      </xs:complexContent>
+    </xs:complexType>
+  </xs:element>
+  <xs:element name="redesignate" type="anyType"/>
+  <xs:element name="redesignate-para">
+    <xs:complexType>
+      <xs:attributeGroup ref="lib:targetAttributes"/>
+      <xs:attributeGroup ref="sharedAttributes"/>
+      <xs:attribute name="num-value" type="xs:string" use="optional"/>
+    </xs:complexType>
+  </xs:element>
+  <xs:element name="recodify-section">
+    <xs:complexType>
+      <xs:attributeGroup ref="lib:targetAttributes"/>
+      <xs:attributeGroup ref="sharedAttributes"/>
+      <xs:attribute name="after" type="xs:string" use="optional"/>
+      <xs:attribute name="before" type="xs:string" use="optional"/>
+      <xs:attribute name="num-value" type="xs:string" use="optional"/>
+    </xs:complexType>
+  </xs:element>
+  <xs:element name="remove" type="anyType"/>
+  <xs:element name="repeal">
+    <xs:complexType>
+      <xs:attributeGroup ref="lib:targetAttributes"/>
+      <xs:attributeGroup ref="sharedAttributes"/>
+      <xs:attribute name="technical" type="xs:string" use="optional"/>
+    </xs:complexType>
+  </xs:element>
+  <xs:element name="replace" type="anyType"/>
+  <xs:element name="street-designation-anno">
+    <xs:complexType>
+      <xs:attributeGroup ref="sharedAttributes"/>
+      <xs:attribute name="action" type="xs:string" use="optional"/>
+    </xs:complexType>
+  </xs:element>
+  <xs:element name="temporary">
+    <xs:complexType>
+      <xs:attributeGroup ref="lib:targetAttributes"/>
+      <xs:attributeGroup ref="sharedAttributes"/>
+      <xs:attribute name="action" type="xs:string" use="optional"/>
+    </xs:complexType>
+  </xs:element>
   <xs:element name="any">
     <xs:complexType>
       <xs:sequence maxOccurs="unbounded">
         <xs:any processContents="lax"/>
       </xs:sequence>
     </xs:complexType>
   </xs:element>
+  <!-- types -->
+  <xs:complexType name="anyType">
+    <xs:complexContent>
+      <xs:extension base="xs:anyType">
+        <xs:attributeGroup ref="sharedAttributes"/>
+      </xs:extension>
+    </xs:complexContent>
+  </xs:complexType>
+  <xs:simpleType name="falseOnlyType">
+    <xs:restriction base="xs:string">
+      <xs:pattern value="false"/>
+    </xs:restriction>
+  </xs:simpleType>
+  <xs:complexType name="findReplaceType">
```

Comment:
It might make sense to do away with the `find` and `replace` attributes - I have no way in XSD 1.0 (which `lxml` and `libxml2` are tied to) of requiring _both_ attributes if it is an empty element, but _not_ allowing the attributes if it has the `find` and `replace` elements.

Another option is to have a `<codify:simple-find-replace>` and a `<codify:full-find-replace>` or some other variation.

Neither option is elegant, unfortunately.